tracker-neuralnet/ftnoir_tracker_neuralnet.cpp

```diff
@@ -0,0 +1,917 @@
+    const std::int64_t input_shape[4] = { 1, 1, input_img_height, input_img_width };
+    input_val = Ort::Value::CreateTensor<float>(allocator_info, input_mat.ptr<float>(0), input_mat.total(), input_shape, 4);
+
+    const std::int64_t output_shape[2] = { 1, 5 };
+    output_val = Ort::Value::CreateTensor<float>(allocator_info, results.data(), results.size(), output_shape, 2);
+}
+
+
+std::pair<float, cv::Rect2f> Localizer::run(
+    const cv::Mat &frame)
+{
+    auto p = input_mat.ptr(0);
+
+    cv::resize(frame, scaled_frame, { input_img_width, input_img_height }, 0, 0, cv::INTER_AREA);
+    scaled_frame.convertTo(input_mat, CV_32F, 1./255., -0.5);
+
+    assert (input_mat.ptr(0) == p);
+    assert (!input_mat.empty() && input_mat.isContinuous());
+    assert (input_mat.cols == input_img_width && input_mat.rows == input_img_height);
+
+    const char* input_names[] = {"x"};
+    const char* output_names[] = {"logit_box"};
+
+    //Timer t_; t_.start();
+
+    const auto nt = omp_get_num_threads();
+    omp_set_num_threads(num_threads);
+    session.Run(Ort::RunOptions{nullptr}, input_names, &input_val, 1, output_names, &output_val, 1);
+    omp_set_num_threads(nt);
+
+    //qDebug() << "localizer: " << t_.elapsed_ms() << " ms\n";
+
+    const cv::Rect2f roi = unnormalize(cv::Rect2f{
+        results[1],
+        results[2],
+        results[3]-results[1], // Width
+        results[4]-results[2] // Height
+    }, frame.rows, frame.cols);
+    const float score = sigmoid(results[0]);
+
+    return { score, roi };
+}
+
+
+PoseEstimator::PoseEstimator(Ort::MemoryInfo &allocator_info, Ort::Session &&session) :
+    session{std::move(session)},
+    scaled_frame(input_img_height, input_img_width, CV_8U),
+    input_mat(input_img_height, input_img_width, CV_32F)
+{
+    {
+        const std::int64_t input_shape[4] = { 1, 1, input_img_height, input_img_width };
+        input_val = Ort::Value::CreateTensor<float>(allocator_info, input_mat.ptr<float>(0), input_mat.total(), input_shape, 4);
+    }
+
+    {
+        const std::int64_t output_shape[2] = { 1, 3 };
+        output_val[0] = Ort::Value::CreateTensor<float>(
+            allocator_info, &output_coord[0], output_coord.rows, output_shape, 2);
+    }
+
+    {
+        const std::int64_t output_shape[2] = { 1, 4 };
+        output_val[1] = Ort::Value::CreateTensor<float>(
+            allocator_info, &output_quat[0], output_quat.rows, output_shape, 2);
+    }
+
+    {
+        const std::int64_t output_shape[2] = { 1, 4 };
+        output_val[2] = Ort::Value::CreateTensor<float>(
+            allocator_info, &output_box[0], output_box.rows, output_shape, 2);
+    }
+}
+
+
+int PoseEstimator::find_input_intensity_90_pct_quantile() const
+{
+    const int channels[] = { 0 };
+    const int hist_size[] = { 255 };
+    float range[] = { 0, 256 };
+    const float* ranges[] = { range };
+    cv::Mat hist; // TODO: Preallocate, but whatever I tried, I ran into asserts in opencv ...
```

Comment:
I've had the same problem, the solution was adding a member variable and letting opencv allocate it the first time. The array is always the same size and type regardless of user-specified options.

Reply:
Hm, doesn't work for me. But doesn't matter. I guess trying anything overly clever here amounts to premature optimization anyway.